draft-moran-suit-mti-algorithms.cddl

```diff
@@ -0,0 +1,127 @@
+SUIT_COSE_tool_tweak /= suit-sha256-hmac-a128-ccm
+SUIT_COSE_tool_tweak /= suit-sha256-es256-hpke-a128gcm
+SUIT_COSE_tool_tweak /= suit-sha256-eddsa-hpke-a128gcm
+SUIT_COSE_tool_tweak /= suit-sha256-hsslms-hpke-a128gcm
+
+SUIT_COSE_tool_tweak /= SUIT_COSE_Profile_HMAC_A128CCM
+SUIT_COSE_tool_tweak /= SUIT_COSE_Profile_ES256_HPKE_A128GCM
+SUIT_COSE_tool_tweak /= SUIT_COSE_Profile_EDDSA_HPKE_A128GCM
+SUIT_COSE_tool_tweak /= SUIT_COSE_Profile_HSSLMS_HPKE_A128GCM
+
+suit-sha256-hmac-a128-ccm = [5, 12]
+suit-sha256-es256-hpke-a128gcm = [-7, 1]
+suit-sha256-eddsa-hpke-a128gcm = [-8, 1]
+suit-sha256-hsslms-hpke-a128gcm = [-46, 1]
+
+SUIT_COSE_Profile_HMAC_A128CCM = SUIT_COSE_Profile<5,12> .and COSE_Messages
+SUIT_COSE_Profile_ES256_HPKE_A128GCM = SUIT_COSE_Profile<-7,1> .and COSE_Messages
+SUIT_COSE_Profile_EDDSA_HPKE_A128GCM = SUIT_COSE_Profile<-8,1> .and COSE_Messages
+SUIT_COSE_Profile_HSSLMS_HPKE_A128GCM = SUIT_COSE_Profile<-46,1> .and COSE_Messages
+
+SUIT_COSE_Profile<authid, encid> = SUIT_COSE_Messages<authid,encid>
+
+SUIT_COSE_Messages<authid, encid> /= SUIT_COSE_Untagged_Message<authid, encid> 
+SUIT_COSE_Messages<authid, encid> /= SUIT_COSE_Tagged_Message<authid, encid> 
```

Comment:
So all profiles must support both tagged and untagged messages?  Why?
I think TEEP would want just one of the two, for a given SUIT profile like SUIT_COSE_Profile_ES256_HPKE_A128GCM.

Reply:
SUIT is currently using tagged signatures because they could be either MAC or Sign. It is also using tagged COSE_Encrypt. Does TEEP need un-tagged? If you want a TEEP-specific schema that has only un-tagged messages, maybe you could do:

```
TEEP_Profile_ES256_HPKE_A128GCM = SUIT_COSE_Profile_ES256_HPKE_A128GCM .and [*any]
```

Reply:
TEEP just needs to specify one of the two (tagged or untagged).  I think it's fine to specify only tagged.